eslintPlugin/src/noAnimatedStyleToNonAnimatedComponent.ts

```diff
@@ -0,0 +1,176 @@
+              ├───────────┼───────┼───────┼───────┼─────────────┤
+              │ Token     │ idx-1 │ idx   │ idx+1 │ idx+2       │
+              ╰───────────┴───────┴───────┴───────┴─────────────╯
+             */
+            if (tokensBefore[idx + 2].value === expectedToken) {
+              isAnimated = true;
+            }
+          });
+          return isAnimated;
+        }
+
+        function checkIdentifierNodeForBeingAnimated(
+          styleExpression: TSESTree.Identifier
+        ) {
+          const variableName = styleExpression.name;
+
+          const isAnimatedStyle = isVariableDefinedAs(
+            variableName,
+            'useAnimatedStyle'
+          );
+
+          if (isAnimatedStyle) {
+            context.report({
+              node,
+              messageId: 'animatedStyle',
+              data: { componentName, variableName },
+            });
+          }
+        }
+
+        function checkObjectNodeForBeingAnimated(
+          styleExpression: TSESTree.ObjectExpression
+        ) {
+          const properties = styleExpression.properties;
+          properties.forEach((property) => {
+            if (property.type === AST_NODE_TYPES.SpreadElement) {
+              return; //Ignore spread elements
+            }
+
+            if (property.value.type === 'Identifier') {
+              const variableName = property.value.name;
+              if (isVariableDefinedAs(variableName, 'useSharedValue')) {
+                const propertyName =
+                  'name' in property.key ? property.key.name : variableName;
+                context.report({
+                  node,
+                  messageId: 'sharedValue',
+                  data: {
+                    propertyName,
+                    propertyValue: variableName,
+                    componentName,
+                  },
+                });
+              }
+            }
+          });
+        }
+
+        function checkArrayNodeForBeingAnimated(
+          styleExpression: TSESTree.ArrayExpression
+        ) {
+          const arrayNodes = styleExpression.elements;
+          arrayNodes.forEach((node) => {
+            if (node?.type === 'Identifier') {
+              checkIdentifierNodeForBeingAnimated(node);
+            } else if (node?.type === 'ArrayExpression') {
+              checkArrayNodeForBeingAnimated(node);
+            } else if (node?.type === 'ObjectExpression') {
+              checkObjectNodeForBeingAnimated(node);
+            }
+          });
+        }
+      },
+    };
+  },
+  meta: {
+    docs: {
+      recommended: 'recommended',
+      description:
+        "Don't pass a reanimated animated style into a non-animated component.",
+    },
+    messages: {
+      sharedValue:
+        "Property  '{{propertyName}}: {{propertyValue}}' is using a sharedValue '{{propertyValue}}', but was used in a default component. Replace your {{componentName}} with an Animated.{{componentName}}",
```

Comment:
Given component doesn't have to be in `Animated` namespace and we shouldn't be that specific here.
```suggestion
        "Property  '{{propertyName}}: {{propertyValue}}' is using a shared value '{{propertyValue}}', but was used in a default component. Replace {{componentName}} with an animated component from Reanimated.",
```